### ISSUE TITLE
Returned categories list differs from categories.yml specification

### ISSUE THREAD

**Issue by aoles** [GIScience/openpoiservice#36]
Categories list returned by the following query to openruteservice pois API differs in structure from the [definition](https://github.com/GIScience/openpoiservice/blob/master/openpoiservice/server/categories/categories.yml). For example, `animals: children: amenity` and `animals: children: shop`  contain 103 and 123 (!) elements, respectively.

```
curl -X POST https://api.openrouteservice.org/pois?api_key=<ORS_API_KEY> \
  -H 'content-type: application/json' -d '{"request": "list"}' > categories_json.txt
```
**OUTPUT: [categories_json.txt](https://github.com/GIScience/openpoiservice/files/2086194/categories_json.txt)**  **[:mag: VIEW ONLINE](https://codebeautify.org/jsonviewer/cb830dc5)**

**Comment by aoles** [comment]
If the intention is to return a 1:1 json version of the yaml definition consider direct conversion similar to https://gist.github.com/mboersma/1329669.

**Comment by TimMcCauley** [comment]
The intention here is not to return a 1:1 json version: what happens is that all tags of a category `amenity` for example are grouped together into one parent with their corresponding IDs which makes it very easy to find while looping over the pbf files. What we could do is add the information of the "top-level" group which has got nothing to do with OSM itself, it's just trying to create a "topic", such as `arts_and_culture` or `facilities` - I will create a separate issue for this.

**Comment by aoles** [comment]
Thanks @TimMcCauley for getting back to me on this.

I get that the grouping to different categories is subjective and not based on OSM. I'm not quite sure, though, that I get the structure of the result of `request=list`. If it's not supposed to return a tag grouping as defined by [https://github.com/GIScience/openpoiservice/blob/master/openpoiservice/server/categories/categories.yml](categories.yml) then what's it's actual logic then?

Did you have a look at `animals: children: amenity` on https://codebeautify.org/jsonviewer/cb830dc5? The grouping of all these amenities under a common `animals` tag puzzles me, I can hardly believe this was meant this way.

**Comment by TimMcCauley** [comment]
Thanks for pointing it out again - I know understand and solved the issue in fde0548c513f250a5ee6c211d9de04643f691ba2
